warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/partition/MultiTableRangePartitioner.java

```diff
@@ -58,15 +60,21 @@ private void readCacheFilesIfNecessary() {
                 // We need a replacement that isn't deprecated, but context.getCacheFiles() returns paths that are not local
                 // No Hadoop documentation seems to indicate what is the correct replacement for this method
                 localCacheFiles = context.getLocalCacheFiles();
-            } catch (IOException e) {
+            } catch (Exception e) {
                 log.error("Failed to get localCacheFiles from context", e);
                 throw new RuntimeException("Failed to get localCacheFiles from context", e);
             }
             
             try {
-                NonShardedSplitsFile.Reader reader = new NonShardedSplitsFile.Reader(context.getConfiguration(), localCacheFiles, isTrimmed());
+                NonShardedSplitsFile.Reader reader = new NonShardedSplitsFile.Reader(context.getConfiguration(), localCacheFiles, getSplitsFileType());
+                if (getSplitsFileType().equals(SplitsFileType.SPLITSANDLOCATIONS)) {
```

Comment:
splitToLocationMap is only used in SortedTabletLocationPartitioner yet the field exists here and the code for initializing it is here.  Perhaps it would be worthwhile to move the field and initialization code to that partitioner by defining a new updateSplits method and calling it here, passing the reader to it.  It could then be overridden by subclasses.

It's just a bit confusing as this partitioner does not use it.  Maybe it's not worth the suggested refactor.

Reply:
Yes, I found the inheritance structure of this puzzling.  I think the original idea was trying to limit duplicated code and also handle a theoretical case where we have several nonsharded tables configured to use different partitioners instead of all of them using the default, but resulting in the parent having to be aware of the type of file that its children need instead of the children simply overriding this method *shrug* I'm speculating.  I've noticed some odd logic in the way we separate the duties of the NonShardedSplitsFile and the ShardedTableMapFile, so I'm working a refactor where we have one SplitsFile to rule them all and each paritioner simply asks for what it needs from that cache and I plan to address this further weirdness there.